assets/javascripts/app/serviceworker.coffee

```diff
@@ -0,0 +1,47 @@
+class app.ServiceWorker
+  $.extend @prototype, Events
+
+  @isEnabled: ->
+    !!navigator.serviceWorker and app.config.service_worker_enabled
+
+  constructor: ->
+    @registration = null
+    @notifyUpdate = true
+
+    navigator.serviceWorker.register(app.config.service_worker_path, {scope: '/'})
+      .then((registration) => @updateRegistration(registration))
+      .catch((error) -> console.error 'Could not register service worker:', error)
```

Comment:
Assuming `@updateRegistration` won’t throw, this would have the same behavior. Not sure if it’s clearer or not:

```coffee
navigator.serviceWorker.register(app.config.service_worker_path, {scope: '/'})
   .then(
      (registration) => @updateRegistration(registration),
      (error) -> console.error('Could not register service worker:', error)
    )
```